src/dotnet/AzureAI.Proxy/ReverseProxy/ThrottlingHealthPolicy.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Yarp.ReverseProxy.Health;
+using Yarp.ReverseProxy.Model;
+
+namespace AzureAI.Proxy.ReverseProxy;
+
+public class ThrottlingHealthPolicy : IPassiveHealthCheckPolicy
+{
+    public static string ThrottlingPolicyName = "ThrottlingPolicy";
+    private readonly IDestinationHealthUpdater _healthUpdater;
+
+    public ThrottlingHealthPolicy(IDestinationHealthUpdater healthUpdater)
+    {
+        _healthUpdater = healthUpdater;
+    }
+
+    public string Name => ThrottlingPolicyName;
+
+    public void RequestProxied(HttpContext context, ClusterState cluster, DestinationState destination)
+    {
+        var headers = context.Response.Headers;
```

Comment:
I will have a look if we need to support extra headers for a PTU deployment, since it has additional utilization headers. Not a blocker for this PR.

Reply:
let's check that. I don't have a PTU to test ;-)